src/viking.js

```diff
@@ -1,11 +1,121 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        let isAlive = this.health > 0;
+        if (isAlive) return `${this.name} has received ${damage} points of damage`
+        return `${this.name} has died in act of combat`
+    }
+
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    receiveDamage(damage) {
+        this.health -= damage;
+        let isAlive = this.health > 0;
+        if (isAlive) return `A Saxon has received ${damage} points of damage`;
+        return `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+
+class War {
+    constructor() {
+        this.vikingArmy = []
+        this.saxonArmy = []
+    }
+        
+    addSaxon(Saxon) {
+        this.saxonArmy.push(Saxon);
+    }
+
+     addViking(Viking) {
+        this.vikingArmy.push(Viking)
+     }
+    
+    vikingAttack() {
+        let randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+        let randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+        let saxonDamage = randomSaxon.receiveDamage(randomViking.strength);
+        if (randomSaxon.health <= 0) this.saxonArmy.splice(randomSaxon, 1);
```

Comment:
el método splice() realmente debería tomar como 1er argumento el índice donde se comienza a cambiar el array (y el segundo el número de elementos a eliminar del array antiguo); aquí randomSaxon es un objeto, para conseguir el índice sería: `this.saxonArmy.indexOf(randomSaxon)`